proj/src/main/webapp/chat.js

```diff
@@ -634,6 +639,39 @@ function friendRequestButton(uid) {
     });
 }
 
+function blockButton(uid) {
+    const currUserRef = firebase.database().ref('/users/'+currentUID+'/blocked/'+uid);
+    const button = document.getElementById('block');
+
+    currUserRef.off();
+    currUserRef.on('value', function(snap) {
```

Comment:
Looking at this again, it looks like the watch on this database reference hangs around even after you change close this window.  It is closed and reopened when you open this window. If you're looking at a lot of users profiles, you will get a build up of watchers which could slow the browser down. Is there a way to call currUserRef.off(); when the profile panel closes?

Reply:
fixed